### PR TITLE
Fix broken neutrino tests

### PR DESCRIPTION
**Pull request by aldebap**
The following things were added/changed/fixed:

. compilation errors fixed in the following tests: neutrino/headerfs/index_test.go , neutrino/bamboozle_unit_test.go and neutrino/blockmanager_test.go;
. elimination of all test cases from  neutrino/headerfs/index_test.go that doesn't make sense anymore;
. fix some segmentation faults in 2 test cases;
. fix in the BatchManager loopBreak detection to avoid logging messages that are not in fact some sort of error;
. refactory of TestBlockManagerDetectBadPeers() test case due to wrong scenario implementation;
. I kept some debug traces used during the debugging of the test code;

### CODE REVIEW

File: neutrino/sync_test.go [hunk 1]
```diff
@@ -957,26 +958,36 @@ func testRandomBlocks(harness *neutrinoHarness, t *testing.T) {
 				return
 			}
 			// Get current basic filter header from the database.
-			curHeader, err := harness.svc.RegFilterHeaders.
-				FetchHeader(&blockHash)
+			//curHeader, err := harness.svc.RegFilterHeaders.
+			//	FetchHeader(&blockHash)
+			curHeader, _, err := harness.svc.NeutrinoDB.FetchBlockHeader(&blockHash)
```
Comment (cjdelisle) [inline neutrino/sync_test.go:963]: Replacing RegFilterHeaders.FetchHeader() with FetchBlockHeader() ?

File: neutrino/sync_test.go [hunk 2]
```diff
@@ -947,8 +947,9 @@ func testRandomBlocks(harness *neutrinoHarness, t *testing.T) {
 				return
 			}
 			// Get previous basic filter header from the database.
-			prevHeader, err := harness.svc.RegFilterHeaders.
-				FetchHeader(&blockHeader.PrevBlock)
+			//prevHeader, err := harness.svc.RegFilterHeaders.
+			//	FetchHeader(&blockHeader.PrevBlock)
+			prevHeader, _, err := harness.svc.NeutrinoDB.FetchBlockHeader(&blockHeader.PrevBlock)
```
Comment (cjdelisle) [inline neutrino/sync_test.go:952]: RegFilterHeaders.FetchHeader() --> FetchBlockHeader() ?

File: neutrino/sync_test.go (hunk 1 above)
Comment (aldebap) [reply neutrino/sync_test.go]: I made a new commit to adjust this: https://github.com/pkt-cash/pktd/pull/327/commits/726c52b4f8c5fdee8a5bab986c605bb6e8986481

File: neutrino/sync_test.go (hunk 2 above)
Comment (aldebap) [reply neutrino/sync_test.go]: I made a new commit to adjust this: https://github.com/pkt-cash/pktd/pull/327/commits/726c52b4f8c5fdee8a5bab986c605bb6e8986481

**Pull request merged**